drivers/flash/flash_stm32.h

```diff
@@ -30,6 +30,17 @@ struct flash_stm32_priv {
 	struct k_sem sem;
 };
 
+#if defined(CONFIG_SOC_SERIES_STM32F4X) || defined(CONFIG_SOC_SERIES_STM32L1X)
+#if DT_PROP(DT_INST(0, st_stm32f4_nv_flash), write_block_size)
+#define FLASH_STM32_WRITE_BLOCK_SIZE \
+	DT_PROP(DT_INST(0, st_stm32f4_nv_flash), write_block_size)
+#endif
+#elif defined(CONFIG_SOC_SERIES_STM32L0X)
+#if DT_PROP(DT_INST(0, st_stm32l0_nv_flash), write_block_size)
+#define FLASH_STM32_WRITE_BLOCK_SIZE \
+	DT_PROP(DT_INST(0, st_stm32l0_nv_flash), write_block_size)
+#endif
+#else
 #if DT_PROP(DT_INST(0, soc_nv_flash), write_block_size)
 #define FLASH_STM32_WRITE_BLOCK_SIZE \
 	DT_PROP(DT_INST(0, soc_nv_flash), write_block_size)
```

Comment:
Do we actually need all these ?

In the end they all generate to the same string:
```
#define DT_N_INST_0_st_stm32_nv_flash DT_N_S_soc_S_flash_controller_58004000_S_flash_8000000
#define DT_N_INST_0_soc_nv_flash      DT_N_S_soc_S_flash_controller_58004000_S_flash_8000000
#define DT_N_NODELABEL_flash0         DT_N_S_soc_S_flash_controller_58004000_S_flash_8000000
```
And write block size ends up in : 
```
#define DT_N_S_soc_S_flash_controller_58004000_S_flash_8000000_P_write_block_size 8
```

The interest of dt binding here is to control the possible value of the property at build time, but in the code `soc_nv_flash` remains valid.

Reply:
I thought these lines were needed to select the corresponding devicetree binding. But now I realize that the `DT_INST()` macro will always match the `soc-nv-flash` compatible in these cases and the end result will be the same.

Fixed in the latest push.